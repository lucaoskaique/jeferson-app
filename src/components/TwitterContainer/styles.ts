import styled from 'styled-components'

export const Wrapper = styled.div`
  max-width: 67rem;
  max-height: 60rem;
  overflow: auto;
  margin: '2rem auto';
`
