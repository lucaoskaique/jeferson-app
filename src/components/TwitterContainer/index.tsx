import { useEffect } from 'react'

import * as S from './styles'

const TwitterContainer = () => {
  useEffect(() => {
    const script = document.createElement('script')
    script.src = 'https://platform.twitter.com/widgets.js'
    document.getElementsByClassName('twitter-embed')[0].appendChild(script)
  }, [])
  return (
    <S.Wrapper>
      <section className="twitterContainer">
        <div className="twitter-embed">
          <a
            className="twitter-timeline"
            data-theme="dark"
            data-tweet-limit="3"
            data-chrome="noheader nofooter noborders"
            href="https://twitter.com/depjeferson"
          >
            Tweets by HeyMarkKop
          </a>
        </div>
      </section>
    </S.Wrapper>
  )
}

export default TwitterContainer
