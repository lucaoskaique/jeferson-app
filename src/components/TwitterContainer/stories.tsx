import { Story, Meta } from '@storybook/react'

import TwitterContainer from '.'

export default {
  title: 'TwitterContainer',
  component: TwitterContainer
} as Meta

export const Default: Story = () => <TwitterContainer />
