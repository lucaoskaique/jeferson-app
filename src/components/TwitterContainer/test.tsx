import { renderWithTheme } from 'utils/tests/helpers'

import TwitterContainer from '.'

describe('<TwitterContainer />', () => {
  it('should render twitter container', () => {
    const { container } = renderWithTheme(<TwitterContainer />)

    expect(container.firstChild).toHaveStyle({
      'max-width': '67rem',
      'max-height': '60rem',
      overflow: 'auto'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
