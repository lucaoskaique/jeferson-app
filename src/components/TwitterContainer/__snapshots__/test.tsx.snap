// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwitterContainer /> should render twitter container 1`] = `
.c0 {
  max-width: 67rem;
  max-height: 60rem;
  overflow: auto;
  margin: '2rem auto';
}

<div
  class="c0"
>
  <section
    class="twitterContainer"
  >
    <div
      class="twitter-embed"
    >
      <a
        class="twitter-timeline"
        data-chrome="noheader nofooter noborders"
        data-theme="dark"
        data-tweet-limit="3"
        href="https://twitter.com/depjeferson"
      >
        Tweets by HeyMarkKop
      </a>
      <script
        src="https://platform.twitter.com/widgets.js"
      />
    </div>
  </section>
</div>
`
